src/arctic3d/modules/pdb.py

```diff
@@ -418,6 +427,151 @@ def fetch_pdb_files(pdb_to_fetch, uniprot_id):
     return validated_pdb_and_cifs
 
 
+def write_renumbered_pdb(pdb_torenum, pdb_aln_string, uniprot_aln_string):
+    """
+    Writes down the renumbered pdb file based on the alignment between the
+     pdb and the uniprot sequence.
+
+    Insertions in the pdb sequence are marked with a letter in the insertion
+    code column (column 27).
+
+    Parameters
+    ----------
+    pdb_torenum : str or Path
+        input pdb file
+
+    pdb_renum : str or Path
+        output pdb file
+
+    pdb_aln_string : str
+        pdb sequence in the alignment
+
+    uniprot_aln_string : str
+        uniprot sequence in the alignment
+    """
+    pdb_renum = Path(f"{pdb_torenum.stem}-renum.pdb")
+    resid_idx, prev_resid, uniprot_resid = -1, -1, 1
+    uniprot_letter_idx = 0
+
+    file_content = ""
+    with open(pdb_renum, "w") as wfile:
+        with open(pdb_torenum) as rfile:
+            for ln in rfile:
```

Comment:
There's too much identantion here, see if you can reduce it. You should be able to do things such as: ` with open(pdb_renum, "w") as wfile, with open(pdb_torenum) as rfile:` and instead of doing the `if` check and going into the block you can do it the other way around and `continue` if it does not pass the check